src/snapshots.js

```diff
@@ -199,7 +199,9 @@ export async function* takeStorybookSnapshots(percy, callback, { baseUrl, flags
     while (snapshots.length) {
       try {
         // use a single page to capture story snapshots without reloading
-        yield* withPage(percy, previewUrl, async function*(page) {
+        // loading an existing story instead of just iframe.html as that triggers `storyMissing` event
+        // This in turn leads to promise rejection and failure
+        yield* withPage(percy, `${previewUrl}?id=${snapshots[0].id}&viewMode=story`, async function*(page) {
```

Comment:
can you refactor this bit and pull out `snapshots[0]` out its used inside as well.

Also one question, 
* why this was not an issue till now?

Reply:
I am not 100% sure about this.
My guess is, there are some additional functions in Storybook v7 which were not present in v6.
Example - https://github.com/storybookjs/storybook/blob/v7.5.0/code/lib/preview-api/src/modules/preview-web/PreviewWithSelection.tsx#L491
This file is not present in `v6.5.16`

Reply:
Alright, let me check the details, not a blocker. 

can you address other comments.

>can you refactor this bit and pull out snapshots[0] out its used inside as well.

> evalSetCurrentStory we use this already to change the story, can you test with multiple story examples (mostly test case should have this scenario, just confirm)

Reply:
> can you refactor this bit and pull out snapshots[0] out its used inside as well.

This wouldn't be very beneficial, as I will still need to extract `id` and other parameters from `snapshots[0]` over [here](https://github.com/percy/percy-storybook/blob/master/src/snapshots.js#L208), since it is inside a while loop, and can have different `snapshots[0]` than what `withPage` is called with.

> evalSetCurrentStory we use this already to change the story, can you test with multiple story examples (mostly test case should have this scenario, just confirm)

Tested locally and verified.

Reply:
> This wouldn't be very beneficial, as I will still need to extract id and other parameters from snapshots[0] over [here](https://github.com/percy/percy-storybook/blob/master/src/snapshots.js#L208), since it is inside a while loop, and can have different snapshots[0] than what withPage is called with.


that's not correct, outer while loop is only meant if the page crashes or something, and we're iterating on the same snapshot array so anyway inner while loop will consume all the snapshot array.

not a blocker for this release, we can maybe refactor in future PRs.